src/markov/markov_approx.jl

```diff
@@ -214,3 +218,402 @@ function estimate_mc_discrete{T}(X::Vector{T})
+    # Check that scaling factor is positive
+    scalingFactor>0 || error("scalingFactor must be a positive number")
+
+    Y = (X-mu)/scalingFactor # standardized grid
+    T = Y'.^collect(1:nMoments)
+end
+
+"""
+Compute the maximum entropy objective function used in discrete_approximation
+    ```julia
+    obj = entropy_obj(lambda,Tx,TBar,q)
+    ```
+##### Arguments
+- `lambda::Vector` : (L x 1) vector of values of the dual problem variables
+- `Tx::Matrix` : (L x N) matrix of moments evaluated at the grid points
+         specified in discrete_approximation
+- `TBar::Vector` : (L x 1) vector of moments of the underlying distribution
+           which should be matched
+- `q::Matrix` : (1 X N) vector of prior weights for each point in the grid.
+
+##### Returns
+- `obj` : scalar value of objective function evaluated at lambda
+"""
+function entropy_obj(lambda::Vector, Tx::Matrix, TBar::Vector, q::Matrix)
+
+    # Some error checking
+    L, N = size(Tx)
+
+    !(length(lambda) != L || length(TBar) != L || length(q) != N) || error("Dimensions of inputs are not compatible.")
+
+    # Compute objective function
+    Tdiff = Tx-repmat(TBar, 1, N)
+    temp = q.*exp.(lambda'*Tdiff)
+    obj = sum(temp)
+    return obj
+end
+
+"""
+Compute gradient of objective function
+##### Returns
+- `grad` : (L x 1) gradient vector of the objective function evaluated
+           at lambda
+"""
+function entropy_grad!(lambda::Vector, grad::Vector, Tx::Matrix, TBar::Vector, q::Matrix)
+    L, N = size(Tx)
+    !(length(lambda) != L || length(TBar) != L || length(q) != N) || error("Dimensions of inputs are not compatible.")
+    Tdiff = Tx-repmat(TBar, 1, N)
+    temp = q.*exp.(lambda'*Tdiff)
+    obj = sum(temp)
+    temp2 = temp.*Tdiff
+    grad .= vec(sum(temp2, 2))
+end
+
+"""
+Compute hessian of objective function
+##### Returns
+- `hess` : (L x L) hessian matrix of the objective function evaluated at `lambda`
+"""
+function entropy_hess!(lambda::Vector, hess::Matrix, Tx::Matrix, TBar::Vector, q::Matrix)
+    L, N = size(Tx)
+    !(length(lambda) != L || length(TBar) != L || length(q) != N) || error("Dimensions of inputs are not compatible.")
+    Tdiff = Tx-repmat(TBar,1,N)
+    temp = q.*exp.(lambda'*Tdiff)
+    temp2 = temp.*Tdiff
+    hess .= temp2*Tdiff'
+end
+
+"""
+find a unitary matrix `U` such that the diagonal components of `U'*AU` is as
+close to a multiple of identity matrix as possible
+##### Arguments
+- `A::Matrix` : square matrix 
+##### Returns
+- `U` : unitary matrix
+- `fval` : minimum value
+"""
+function  min_var_trace(A::Matrix)
+    ==(size(A)...) || throw(ArgumentError("input matrix must be square"))
+
+    K = size(A, 1) # size of A
+    d = trace(A)/K # diagonal of U'*A*U should be closest to d
+    function obj(X, grad)
+        X = reshape(X, K, K)
+        return (norm(diag(X'*A*X)-d))
+    end
+    function unitaryConstraint(res, X, grad)
+        X = reshape(X, K, K)
+        res .= vec(X'*X - eye(Int64(sqrt(length(X)))))
```

Comment:
Does this work on 0.5? It doesn't work on 0.6.

`Int64(sqrt(length(X)))` should either be `round(Int, sqrt(length(X)))`, `floor(Int, sqrt(length(X)))`, or `ceil(Int, sqrt(length(X)))` depending on the behavior you want.